src/GregClient/PackageManagerClient.cs

```diff
@@ -0,0 +1,144 @@
+﻿using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Greg.Requests;
+using Greg.Responses;
+using System.Net;
+using System.IO;
+using Greg.Utility;
+using Newtonsoft.Json;
+
+namespace Greg
+{
+    /// <summary>
+    /// Package Manager REST Client for ACG API
+    /// </summary>
+    public class PackageManagerClient : IGregClient
+    {
+        private const string AFC_CODE = "DY1ON1";
+        private readonly RestClient _client;
+        private readonly RestClient _fileClient;
+
+        public string BaseUrl { get { return _client.BaseUrl.ToString(); } }
+
+        public readonly IAuthProvider _authProvider;
+        public IAuthProvider AuthProvider
+        {
+            get { return _authProvider; }
+        }
+
+        public PackageManagerClient(IAuthProvider provider, string packageManagerUrl, string fileStorageUrl)
+        {
+            _authProvider = provider;
+            _client = new RestClient(packageManagerUrl);
+            _fileClient = new RestClient(fileStorageUrl);
+        }
+
+        /// <summary>
+        /// For GET request we need to pass AFC(Affiliation Code) in request header.
+        /// For all other request and Member requests we need to pass additional X-Session in request header. 
+        /// If reuest is for upload file _flieClient is used, as URL for ACG API and File Upload server are different.
+        /// </summary>
+        /// <param name="m"></param>
+        /// <returns></returns>
+        private IRestResponse ExecuteInternal(Request m)
+        {
+            var req = new RestRequest(m.Path, m.HttpMethod);
+
+            m.Build(ref req);
+
+
+            if (m.HttpMethod == Method.GET && !m.Path.Contains("members"))
+                SignRequest(req);
+            else
+                AuthProvider.SignRequest(ref req, _client);
+
+            if (((PackageManagerRequest)(m)).fileRequest)
```

Comment:
What if m is not PackageManagerRequest? when casting will fail, it will throw NullReferenceException.